### PR TITLE
Fix Cypher RETURN clause to unwrap single element results

### PR DESCRIPTION
**Pull request by ExtReMLapin**
## What does this PR do?

This PR fixes the OpenCypher query engine to properly unwrap single element results when returning a single node or edge variable. When a Cypher query like `MATCH (n) RETURN n` is executed, the result should be the element directly (a Vertex or Edge), not wrapped in a projection map like `{"n": element}`. This aligns with standard Cypher behavior and matches the behavior of the Gremlin-based Cypher engine.

### Changes Made:

1. **FinalProjectionStep.java**: Added logic to detect when a single property is being returned that resolves to a Document (vertex/edge) and return it as an element result directly instead of wrapping it in a projection.

2. **CypherResultFormatTest.java** (new): Added comprehensive test coverage for the new behavior, including:
   - Single node returns should be elements
   - Single edge returns should be elements
   - Property access returns should still be projections
   - Multiple column returns should still be projections
   - Aliased single elements should still unwrap

3. **Test Updates**: Updated 15+ existing test files to use `result.toElement()` instead of `result.getProperty("varName")` when accessing single returned elements, reflecting the new behavior.

## Motivation

The previous behavior of wrapping single element returns in projection maps was inconsistent with standard Cypher semantics and the Gremlin-based implementation. This caused confusion and required workarounds in test code. By unwrapping single element results, we provide a more intuitive and standards-compliant API.

## Related issues

This fix addresses the inconsistency between the OpenCypher and Gremlin-based Cypher engines in result formatting.

## Additional Notes

- The change is backward compatible in the sense that `result.toElement()` provides the same functionality as the previous `result.getProperty("varName")` approach
- Multi-column returns and property projections continue to work as before
- All existing tests have been updated to reflect the new behavior
- The implementation correctly handles edge cases like aliased returns and aggregate functions

## Checklist

- [x] I have run the build using `mvn clean package` command
- [x] My unit tests cover both failure and success scenarios

https://claude.ai/code/session_01LmNejBtujWGirVHcc2ackZ

### CODE REVIEW

**Comment by ExtReMLapin** [comment]
Claude wrote all of this, I just described the issue and the diagnosis.

File: engine/src/test/java/com/arcadedb/query/opencypher/CypherResultFormatTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    // MATCH ()-[r]->() RETURN r should return elements directly
+    final ResultSet result = database.query("opencypher", "MATCH ()-[r:KNOWS]->() RETURN r");
+    final List<Result> results = new ArrayList<>();
+    while (result.hasNext()) {
+      results.add(result.next());
+    }
+
+    assertThat(results).hasSize(1);
+    assertThat(results.getFirst().isElement())
+        .as("Edge result should be an element, not a projection").isTrue();
```
[the remark this answers is not in the document]
Comment (ExtReMLapin) [reply engine/src/test/java/com/arcadedb/query/opencypher/CypherResultFormatTest.java:91]: I'm pretty sure we target >= Java 21
